---
title: "11_beta_div_CJS"
author: "Colin"
date: "2025-08-04"
output: html_document
---
# This is the beta diversity analysis code from Aizpurua et al. in prep https://alberdilab.github.io/desert_bats_metagenomics/diversity.html#beta-diversity 

```{r load_data_beta, comment="", message=FALSE, warning=FALSE, eval=TRUE}
load("../data/all_wombats/data.Rdata")

library(rmarkdown)
library(R.utils)
library(knitr)
library(tidyverse)
library(devtools)
library(tinytable)
library(broom)
library(broom.mixed)
library(janitor)

# For tree handling
library(ape)
library(phyloseq)
library(phytools)

# For plotting
library(plotly)
library(ggplot2)
library(ggrepel)
library(ggpubr)
library(ggnewscale)
library(gridExtra)
library(ggtreeExtra)
library(ggtree)
library(ggh4x)

# For statistics
library(spaa)
library(vegan)
library(Rtsne)
library(geiger)
library(hilldiv2)
library(distillR)
library(ANCOMBC)
library(lme4)
library(matrixStats)
```



```{r beta_div, eval=FALSE, message=FALSE, warning=FALSE, comment=""}
beta_q0n <- genome_counts_filt %>%
  select(where(~!all(. == 0))) %>% # remove empty samples
  column_to_rownames(., "genome") %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts_filt %>%
  select(where(~!all(. == 0))) %>% # remove empty samples
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1)

genome_counts_filt <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_tree <- keep.tip(genome_tree, tip=genome_counts_filt$genome)

beta_q1p <- genome_counts_filt %>%
  select(where(~!all(. == 0))) %>% # remove empty samples
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1, tree = genome_tree)

genome_counts_filt <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt <- genome_counts_filt %>%
  remove_rownames() %>% 
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts1 <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt$genome,]
genome_gifts1 <- genome_gifts1[, colSums(genome_gifts1 != 0) > 0]

dist <- genome_gifts1 %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

beta_q1f <- genome_counts_filt %>%
  select(where(~!all(. == 0))) %>% # remove empty samples
  filter(genome %in% labels(dist)[[1]]) %>%
  column_to_rownames(., "genome") %>%
  hillpair(., q = 1, dist = dist)
```



```{r save_beta, comment="", message=FALSE,echo=FALSE,warning=FALSE, eval=FALSE}

save(beta_q0n, 
     beta_q1n, 
     beta_q1p, 
     beta_q1f,
     genome_gifts,
     genome_gifts1,
     genome_tree,
     file = "../data/all_wombats/beta2.Rdata")

```


```{r set_seed, comment="", message=FALSE,echo=FALSE,warning=FALSE, eval=FALSE}
set.seed(1234)
```


```{r permanova_q0n(richness)_S, comment="", message=FALSE, warning=FALSE}

sample_metadata_row <- column_to_rownames(sample_metadata, "sample") 
sample_metadata_row <- sample_metadata_row[labels(beta_q0n$S), ]
# sample_metadata_trim <- sample_metadata %>%
  # filter(sample != "EHI01806" & sample != "EHI01812")

betadisper(beta_q0n$S, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q0n$S ~ host_species, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q0n$S))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_nmds_richness_plot_median, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}

nmds_beta_richness_S <- beta_q0n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )

```

```{r permanova_q1n(neutral), comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1n$S, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q1n$S ~ host_species, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$S))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_nmds_neutral_plot_median, comment="", message=FALSE, warning=FALSE}

nmds_beta_neutral_S <- beta_q1n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )


```

```{r permanova_q1p(phylogenetic), comment="", message=FALSE, warning=FALSE}

betadisper(beta_q1p$S, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE)
adonis2(beta_q1p$S ~ host_species,
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1p$S))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()

```

```{r beta_div_nmds_phylogenetic_plot_median, comment="", message=FALSE, warning=FALSE}

beta_q1p$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0)
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )

```

```{r permanova_q1f(functional), comment="", message=FALSE, warning=FALSE}

betadisper(beta_q1f$S, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE)
adonis2(beta_q1f$S ~ host_species,
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1f$S))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()

```

```{r beta_div_nmds_functional_plot_median, comment="", message=FALSE, warning=FALSE}


nmds_beta_func_S <- beta_q1f$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )

```

# computing other diversity metrics to compare 'U'

```{r permanova_q0n(richness)_U, comment="", message=FALSE, warning=FALSE}

betadisper(beta_q0n$U, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q0n$U ~ host_species, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q0n$U))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()

```

```{r beta_div_nmds_richness_plot_median_U, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE }

nmds_beta_richness_U <- beta_q0n$U %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )
```

```{r permanova_q1n(neutral)_U, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1n$U, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q1n$U ~ host_species, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$U))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()


```

```{r permanova_q1p(neutral)_U, comment="", message=FALSE, warning=FALSE}

 nmds_beta_neutral_U <- beta_q1n$U %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )

```

```{r permanova_q1p(phylogenetic)_U, message=FALSE, warning=FALSE, comment=""}

betadisper(beta_q1p$U, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE)
adonis2(beta_q1p$U ~ host_species,
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1p$U))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_nmds_phylogenetic_plot_median_U, comment="", message=FALSE, warning=FALSE}

nmds_beta_phylo_U <- beta_q1p$U %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )
```

```{r permanova_q1f(functional)_U, comment="", message=FALSE, warning=FALSE}

betadisper(beta_q1f$U, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE)
adonis2(beta_q1f$U ~ host_species,
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1f$U))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_nmds_functional_plot_median_U, comment="", message=FALSE, warning=FALSE}
nmds_beta_func_U <- beta_q1f$U %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )

```

```{r permanova_q0n(richness)_C, comment="", message=FALSE, warning=FALSE}

betadisper(beta_q0n$C, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q0n$C ~ host_species, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q0n$C))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_div_nmds_richness_plot_median_C, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q0n$U %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )
```

```{r permanova_q1n(neutral)_C, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1n$C, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q1n$C ~ host_species, 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$C))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r permanova_q1p(neutral)_C, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}

nmds_beta_neutral_C <- beta_q1n$C %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )
```

```{r permanova_q1p(phylogenetic)_C, message=FALSE, warning=FALSE, comment=""}
betadisper(beta_q1p$C, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE)
adonis2(beta_q1p$C ~ host_species,
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1p$C))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()

```

```{r beta_div_nmds_phylogenetic_plot_median_C, comment="", message=FALSE, warning=FALSE,fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q1p$C %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )

```

```{r permanova_q1f(functional)_C, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1f$C, sample_metadata_row$host_species) %>% permutest(., pairwise = TRUE)
adonis2(beta_q1f$C ~ host_species,
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1f$C))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()

```

```{r beta_div_nmds_functional_plot_median_C, comment="", message=FALSE, warning=FALSE}
beta_q1f$C %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace = 0)
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species)) +
    scale_color_manual(values = c("#e5bd5b", "#6b7398","#e2815a")) +
    scale_shape_manual(values = 1:10) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 20, face = "bold"),
      axis.text = element_text(face = "bold", size = 18),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18),
      legend.position = "right", legend.box = "vertical"
    )
```
# Let's try plotting a PCoA of the hill numbers maybe that will show something different. 
